vnext/Microsoft.ReactNative/Views/TouchEventHandler.cpp

```diff
@@ -437,61 +452,142 @@ bool TouchEventHandler::TagFromOriginalSource(
+    //
+    // The side-effect is that RTL text may have additional valid hit box
+    // space at the end of a line, but it's better than the alternative of the
+    // hit box excluding the last character on a line.
+    const auto prevPointer = textPointer.GetPositionAtOffset(-1, winrt::LogicalDirection::Forward);
+    if (prevPointer != nullptr) {
+      const auto prevRect = prevPointer.GetCharacterRect(winrt::LogicalDirection::Forward);
+      if (rect.Y == prevRect.Y) {
+        right = prevRect.X;
       }
+    }
+  }
 
-      // Approximate the bounding rect (for now, don't account for text wrapping).
-      if ((startRect.X <= pointerPos.X) && (endRect.X + endRect.Width >= pointerPos.X) &&
-          (startRect.Y <= pointerPos.Y) && (endRect.Y + endRect.Height >= pointerPos.Y)) {
-        isHit = true;
-        return nullptr;
-      }
+  // The character is after the point if the Y-coordinate of the point is above
+  // (less than) the top of the character rect, or if the Y-coordinate is above
+  // (less than) the bottom of the character rect and the X-coordinate is
+  // "before" the beginning of the character rect (less than the left side for
+  // LTR or greater than the right for RTL):
+  // ┌────> X
+  // │ (x,y) ┌─────┐  ┌─────┐ (x,y)
+  // ↓ ┌─────┘     │  │     └─────┐
+  // Y └───────────┘  └───────────┘
+  //        LTR            RTL
+  const auto isBeforeX = !isRtl ? point.X < rect.X : point.X > right;
+  return point.Y < rect.Y || (point.Y < bottom && isBeforeX);
+}
+
+bool IsRTL(const winrt::TextPointer &textPointer) {
+  auto currentPointer = textPointer.GetPositionAtOffset(1, winrt::LogicalDirection::Forward);
+  auto currentRect = textPointer.GetCharacterRect(winrt::LogicalDirection::Forward);
+  auto firstCharacterRect = currentRect;
+  while (currentPointer != nullptr) {
+    currentRect = currentPointer.GetCharacterRect(winrt::LogicalDirection::Forward);
+    // If we haven't figured out the direction by the time we reach the end of
+    // a line, try again for the next line.
+    if (currentRect.Y != firstCharacterRect.Y) {
+      firstCharacterRect = currentRect;
+    } else if (currentRect.X != firstCharacterRect.X) {
+      return currentRect.X < firstCharacterRect.X;
+    }
+  }
+
+  // Assume LTR if there are not enough characters to determine LTR vs. RTL.
+  // This may occur for TextBlocks with a single character per line.
+  return false;
+}
+
+winrt::TextPointer TouchEventHandler::GetPositionFromPoint(
+    const winrt::TextBlock &textBlock,
+    const winrt::Point &targetPoint) {
+  // Since characters in a TextBlock are sorted from top-left to bottom-right
+  // (or top-right to bottom-left for RTL), we can use binary search to find
+  // the character with bounds that container the pointer point.
+  //
+  // This algorithm currently makes the following assumptions:
+  // 1. Characters on the same line have the same Rect::Y value
```

Comment:
One more thing to check: will this assumption break down if the app modifies the LineStackingStrategy property: https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.linestackingstrategy?view=winrt-19041

Reply:
#wontfix - the assumption is documented and there are no references to LineStackingStrategy in react-native-windows.